### PR TITLE
Feat/middleware

### PR DESCRIPTION
**Pull request by leisterfrancisco**
Create middle through Dockerfile

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  writer-middleware:
+    container_name: writer-middleware
+    build: ./eosio-writer-middleware-docker/
+    ports:
+      - '8080:8080'
+      # - '8888:8888' 
+    # volumes:
+    #   - ./eosio-writer-middleware/temp/node_modules:/app/node_modules
+    #   - ./eosio-writer-middleware/:/app
```
Comment (leisterfrancisco) [inline docker-compose.yml]: Working on

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+version: '3.5'
+
+services:
+  lacchain-network:
+    container_name: lacchain-network
+    build: ./eosio-lacchain-network
+    ports:
+      - '8888:8888' 
+      # - '8080:8080'
```
Comment (leisterfrancisco) [inline docker-compose.yml]: Working on

File: eosio-lacchain-network/start.sh [hunk 3]
```diff
@@ -47,6 +47,25 @@ setup_accounts() {
...
+          "waits":[]
+      },
+      "owner" : {
+          "threshold":1,
+          "keys":[],
+          "accounts":[{"weight":1, "permission":{"actor":"eosio", "permission":"active"}}],
+          "waits":[]
+      }
+  }' -p eosio
+
```
Comment (leisterfrancisco) [inline eosio-lacchain-network/start.sh:68]: Missing abi file

File: test/test.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
```
Comment (leisterfrancisco) [inline test/test.sh]: Pending final test

Comment (leisterfrancisco) [reply test/test.sh]: @xavier506 let me know if we let this file as test example?

Comment (leisterfrancisco) [reply test/test.sh]: @xavier506 could you help me with this?

**Comment by leisterfrancisco** [comment]
@xavier506 according with [BIOS Boot Sequence](https://developers.eos.io/welcome/latest/tutorials/bios-boot-sequence), cleos default port is 8888.

image

 Would you still want to have 8888 for middleware?

**Comment by xavier506** [comment]
> @xavier506 according with [BIOS Boot Sequence](https://developers.eos.io/welcome/latest/tutorials/bios-boot-sequence), cleos default port is 8888.
> 
> image
> 
> Would you still want to have 8888 for middleware?

@leisterfrancisco  I think its best if the middle ware exposes the same port that the genesis node would , and the genesis node is using a different port behind the middleware

think of the middle ware as the same http service provided by the genesis node

**Comment by leisterfrancisco** [comment]
> > @xavier506 according with [BIOS Boot Sequence](https://developers.eos.io/welcome/latest/tutorials/bios-boot-sequence), cleos default port is 8888.
> > image
> > Would you still want to have 8888 for middleware?
> 
> @leisterfrancisco I think its best if the middle ware exposes the same port that the genesis node would , and the genesis node is using a different port behind the middleware
> 
> think of the middle ware as the same http service provided by the genesis node

@xavier506 ok, I just wanted to make sure you knew that, I'll do that

**Review by xavier506 — APPROVED** [review]
LGTM

File: eosio-lacchain-network/start.sh [hunk 5]
```diff
@@ -47,6 +47,28 @@ setup_accounts() {
     cleos create account eosio $account $TESTNET_EOSIO_PUBLIC_KEY
   done
 
+  echo "====================================== Creating writer account ======================================"
+  cleos push action eosio newaccount \
```
Comment (xavier506) [inline eosio-lacchain-network/start.sh]: cleos wil pint to port `:8888` if url is not specified , 

We should have port `8888` exposed for the middle ware and `8080` for nodeos

File: .env.example [hunk 6]
```diff
@@ -0,0 +1,8 @@
+DOCKER_PASSWORD=secret
+MIDDLEWARE_EOS_API_ENDPOINT=http://lacchain-network:8888
+MIDDLEWARE_EOS_API_CHAIN_ID=5821525c6588037e2b066b992fcac34909a5b7f1ea8d5a393f6720fca3750d61
+MIDDLEWARE_EOS_WRITER_ACCOUNT_NAME=localdev
+MIDDLEWARE_EOS_WRITER_PUB_KEY=EOS1234
+MIDDLEWARE_EOS_WRITER_PRI_KEY=54321
+MIDDLEWARE_SERVER_PORT=8080
```
Comment (xavier506) [inline .env.example]: We should have port `8888` exposed for the middle ware and `8080` for nodeos

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+version: '3.5'
+services:
+  lacchain-network:
+    container_name: lacchain-network
+    build: ./eosio-lacchain-network
+    ports:
+      - '8080:8080'
```
Comment (xavier506) [inline docker-compose.yml:7]: 💯 🏆 🎖️

File: eosio-lacchain-network/config.ini [hunk 8]
```diff
@@ -11,7 +11,7 @@ plugin = eosio::history_api_plugin
 plugin = eosio::http_plugin
 
 # HTTP
-http-server-address = 0.0.0.0:8888
+http-server-address = 0.0.0.0:8080
```
Comment (xavier506) [inline eosio-lacchain-network/config.ini:14]: 👍

File: .env.example [hunk 9]
```diff
@@ -0,0 +1,8 @@
+DOCKER_PASSWORD=secret
+MIDDLEWARE_EOS_API_ENDPOINT=http://lacchain-network:8080
+MIDDLEWARE_EOS_API_CHAIN_ID=5821525c6588037e2b066b992fcac34909a5b7f1ea8d5a393f6720fca3750d61
+MIDDLEWARE_EOS_WRITER_ACCOUNT_NAME=localdev
+MIDDLEWARE_EOS_WRITER_PUB_KEY=EOS1234
+MIDDLEWARE_EOS_WRITER_PRI_KEY=54321
+MIDDLEWARE_SERVER_PORT=8888
```
Comment (xavier506) [inline .env.example:7]: 🥇

**Pull request merged**